### ISSUE TITLE
The post_process() method is skipped if `db_transaction = False`

### ISSUE THREAD

**Issue by peterfarrell** [mixxorz/django-service-objects#51]
If you set `db_transaction = False` on your service and implement a `post_process()` method, the post process method does not execute. While examining the code, it makes sense why because the on transaction commit hook is being used.  However, the behavior is a bit obtuse especially if you are not familiar with the internals of the library.

The solution I propose is if `db_transaction = False` and a `post_process()` is implemented, then automatically execute the `post_process()` in sequence after the `process()`. This would be useful for services where transaction safety is not needed (like an API call, etc.) and would make the feature in parity when `db_transaction = True`.

I do believe it's as easy as implementing a `finally` statement:

```python
    @contextmanager
    def _process_context(self):
        """
        Returns the context for :meth:`process`
        :return:
        """
        if self.db_transaction:
            with transaction.atomic(using=self.using):
                if self.has_on_commit_action:
                    transaction.on_commit(self.post_process)
                yield
        else:
            try:
                yield
            finally:
                if self.has_on_commit_action:
                    self.post_process()
```
Note: I would likely rename `has_on_commit_action` to `has_post_process_action` since it would not be transaction related.

Alternatively, we could throw an exception if you define a post_process when transaction is false however I think making the feature act in parity is better.

Please let me know your thoughts and I'd be happy to make a PR.

**Comment by c17r** [comment]
I don't think `post_process` should run if the main part of the process fails. I see the intent of `post_process` being for things like notifications/syncing/etc which shouldn't fire if things have failed and the database hasn't updated.

But I do think `post_process` should run regardless of the transaction setting as long as an exception doesn't make it all the way to the library level.

**Comment by peterfarrell** [comment]
I realize this morning that the finally would always run would would cause the post_process to always run even if the process encountered an exception.  Thanks for pointing that out @c17r. Let me test something before commenting further.

**Comment by peterfarrell** [comment]
@c17r I tested the following and it just works to append the if statement after the yield.  If an unhandled exception is raised in the `process` the `post_process` is not run.

```python
   @contextmanager
    def _process_context(self):
        """
        Returns the context for :meth:`process`
        :return:
        """
        if self.db_transaction:
            with transaction.atomic(using=self.using):
                if self.has_on_commit_action:
                    transaction.on_commit(self.post_process)
                yield
        else:
            yield
            if self.has_on_commit_action:
                self.post_process()
```

**Comment by peterfarrell** [comment]
@c17r PR made and tests created/updated.

**Comment by c17r** [comment]
Merged in #52, thanks!
